airbyte-integrations/connectors/source-google-sheets/source_google_sheets/config_migrations.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+#
+
+import json
+from typing import Any, List, Mapping
+
+import dpath
+import orjson
+
+from airbyte_cdk.config_observation import create_connector_config_control_message
+from airbyte_cdk.entrypoint import AirbyteEntrypoint
+from airbyte_cdk.models import AirbyteMessageSerializer
+from airbyte_cdk.sources.message import InMemoryMessageRepository, MessageRepository
+from source_google_sheets import SourceGoogleSheets
+
+
+class MigrateServiceAccountInfo:
```

Comment:
I'm a bit concerned about this migration because `credentials. service_account` is not in the spec which means that it is not marked as a `airbyte_secret` and therefore I don't think it'll be obfuscated in the database. Do you know if this is an issue?

Reply:
As a solution I could add `service_account` to spec with `airbyte_hidden` key, but in that case migration will be breaking change as user need update spec.

Reply:
I have added json.loads to the JWT authenticator here - https://github.com/airbytehq/airbyte-python-cdk/pull/301. So, I have removed config migration